src/services/email/libs/data-lib.js

```diff
@@ -0,0 +1,95 @@
+import { DateTime } from "luxon";
+
+import { getLookupValues } from "./lookup-lib";
+
+const formatAttachments = (formatType, attachmentList) => {
+    const formatChoices = {
+        "text": {
+            begin: "\n\n",
+            joiner: "\n",
+            end: "\n\n"
+        },
+        "html": {
+            begin: "<ul><li>",
+            joiner: "</li><li>",
+            end: "</li></ul>"
+        },
+    };
+    const format = formatChoices[formatType];
+    if (!format) {
+        console.log("new format type? ", formatType);
+        return "attachment List";
+    }
+    if (!attachmentList || attachmentList.length === 0)
+        return "no attachments";
+    else
+        return `${format.begin}${attachmentList.map(a => `${a.title}: ${a.filename}`).join(format.joiner)}${format.end}`;
+};
+
+const formatDateFromTimestamp = (timestamp) => {
+    if (!timestamp || timestamp <= 0) return "Pending";
+    return DateTime.fromMillis(timestamp)
+        .toFormat("DDDD");
+
+};
+
+function formatNinetyDaysDate(emailBundle) {
+    if (!emailBundle?.notificationMetadata?.submissionDate) return "Pending";
+    return DateTime.fromMillis(emailBundle.notificationMetadata.submissionDate)
+        .plus({ days: 90 })
+        .toFormat("DDDD '@ 11:59pm ET'");
+
+}
+
+export const buildEmailData = async (bundle, data) => {
+    const returnObject = {};
+
+    const lookupValues = await getLookupValues(bundle.lookupList, data.id);
+    data = { ...data, ...lookupValues };
+
+    if (!bundle.dataList || !Array.isArray(bundle.dataList) || bundle.dataList.length === 0)
+        return { error: "init statement fail", bundle, data };
+
+    bundle.dataList.forEach((dataType) => {
+        switch (dataType) {
```

Comment:
Just a thought, in the future maybe we could make these dataTypes actual typed objects that can be enforced to do their own data transformations via a structured method? Would simplify this area of code and leave each dataType responsible for its own formatting.

Reply:
ok, it is going to take me an hour or two to even understand this....  you java/typescript people with your "fancy overloading" :)

Reply:
but not today, btw, this is a "fix" I will push for later :)